### PR TITLE
New function to convert level_enum from string

### PR DESCRIPTION
**Pull request by fegomes**
## Requeriment
Configure log level using name. (trace, debug, info...)

## Solution
Creation a new function to convert level_enum from string  - **to_level_enum(const char*)** 

## Example
``` cpp
spdlog::level::level_enum lvl = spdlog::level::to_level_enum("debug");
```

### CODE REVIEW

**Comment by gabime** [comment]
Thanks for the pull.  Why is the use case  here?

Also, A lookup table(```unordered_map<string, level>```) should be considered for better performance (but need to be benched - for such small array not user it is faster)

**Comment by fegomes** [comment]
## Example:

``` ini
// log.ini
[General]
path=/home/user/log
pattern=%+
level=debug -- it is more simple using name for users
```

I didn`t make with unordered_map because you maked rule to redefine the SPDLOG_LEVEL_NAMES macro

**Comment by gabime** [comment]
>I didn`t make with unordered_map because you maked rule to redefine the SPDLOG_LEVEL_NAMES macro

How about:
```c++
inline spdlog::level::level_enum from_name(const char* name)
{
	static std::unordered_map < std::string, level_enum> name_to_level
	{
		{ level_names[0] , level::trace },
		{ level_names[1] , level::debug },
		{ level_names[2] , level::info },
		{ level_names[3] , level::warn },
		{ level_names[4] , level::err },
		{ level_names[5] , level::critical },
		{ level_names[6] , level::off }
	};

	auto lvl = name_to_level.find(name);
	return lvl != name_to_level.end() ? lvl->second : level::off;						
}
```

**Comment by fegomes** [comment]
It's so good, but the users can't customize SPDLOG_LEVEL_NAMES macro. How you can see in your code

**spdlog/common.h:85**

``` cpp
#if !defined(SPDLOG_LEVEL_NAMES)
#define SPDLOG_LEVEL_NAMES { "trace", "debug", "info",  "warning", "error", "critical", "off" }
#endif
```

**Comment by fegomes** [comment]
I think this function is simple and I don't imagine call many times during application running, so I preserved the compatibility with your SPDLOG_LEVEL_NAMES macro.

**Comment by gabime** [comment]
> It's so good, but the users can't customize SPDLOG_LEVEL_NAMES macro. How you can see in your code

Why not? I tried and it worked, You need to define SPDLOG_LEVEL_NAMES before including spdlog.h

**Comment by fegomes** [comment]
Yeah it's works, but the function "from_name" or "to_level_name" doesn't works, when user define SPDLOG_LEVEL_NAMES in your code.

### example

``` cpp
#define SPDLOG_LEVEL_NAMES { "customize1", "customize2", "customize3" }
#include <spdlog/spdlog.h>
...
```
Could you create a  protection if user define SPDLOG_LEVEL_NAMES, How about in spdlog.h

``` cpp
// spdlog.h
#pragma once

#ifdef SPDLOG_LEVEL_NAMES 
    #error "Error: You must not define SPDLOG_LEVEL_NAMES" 
#endif
```

File: include/spdlog/common.h [hunk 1]
```diff
@@ -51,6 +51,9 @@ namespace spdlog
 
 class formatter;
 
+template<class T, size_t N>
+constexpr size_t size(T(&)[N]) { return N; }
```
Comment (boimart1) [inline include/spdlog/common.h]: VC 2013 doesn't support `constexpr`

**Comment by fegomes** [comment]
@gabime,  What do you think about commits?

**Comment by gabime** [comment]
@fegomes Looks good. What about the SPDLOG_LEVEL_NAMES  ?

**Comment by fegomes** [comment]
Are you saying about compile protection or name of variable?

**Comment by gabime** [comment]
To be able to handle custom log levels which might be defined in SPDLOG_LEVEL_NAMES

**Comment by fegomes** [comment]
I didn't see benefits to customize **SPDLOG_LEVEL_NAMES**  because your enum level_enum is static.

And more, if user not define 7 values, It's causes runtime error.

**Comment by gabime** [comment]
> I didn't see benefits to customize SPDLOG_LEVEL_NAMES because your enum level_enum is static.

Doesn't matter - we need to decide whether to take into account that someone would use it. Do we want ```from_name``` respect it or not? I think yes

> And more, if user not define 7 values, It's causes runtime error.

This would happen anyway - regardless of the ```from_name``` function

**Comment by fegomes** [comment]
> Doesn't matter - we need to decide whether to take into account that someone would use it. Do we want from_name respect it or not? I think yes

Ok! ;).. I changed the code.

**Pull request merged**